### PR TITLE
Document new status API endpoints

### PR DESCRIPTION
**Pull request by juliusv**
These were added in https://github.com/prometheus/prometheus/pull/6243
for the React UI.

Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by juliusv** [comment]
@brian-brazil (FYI @boyskila)

File: docs/querying/api.md [hunk 1]
```diff
@@ -696,6 +696,66 @@ $ curl http://localhost:9090/api/v1/status/flags
...
+    "goroutineCount": 48,
+    "GOMAXPROCS": 4,
+    "GOGC": "",
+    "GODEBUG": "",
+    "storageRetention": "15d"
+  }
+}
+```
+
+**NOTE**: The exact returned runtime properties may change without notice between Prometheus versions!
```
Comment (brian-brazil) [inline docs/querying/api.md]: I don't think we need the exclamation mark.

File: docs/querying/api.md [hunk 2]
```diff
@@ -696,6 +696,66 @@ $ curl http://localhost:9090/api/v1/status/flags
...
+    "buildUser": "julius@desktop",
+    "buildDate": "20191102-16:19:59",
+    "goVersion": "go1.13.1"
+  }
+}
+```
+
+**NOTE**: The exact returned build properties may change without notice between Prometheus versions!
+
 *New in v2.2*
```
Comment (brian-brazil) [inline docs/querying/api.md]: Needs adding/updating/moving.

Comment (juliusv) [reply docs/querying/api.md]: Hmmm... or how about I just add equivalent `New in v2.14` notices for the new endpoints and keep this as is?

File: docs/querying/api.md (hunk 1 above)
Comment (juliusv) [reply docs/querying/api.md]: I thought the exclamation mark would keep the API stability lawyers away ;) Replaced with a period.

File: docs/querying/api.md (hunk 2 above)
Comment (juliusv) [reply docs/querying/api.md]: Oh, the "new in" referred to the section above it, not below it, right?

Comment (juliusv) [reply docs/querying/api.md]: Hopefully fixed now.

File: docs/querying/api.md (hunk 1 above)
Comment (brian-brazil) [reply docs/querying/api.md]: We need to mention it, not shout it from the rooftops :)

**Comment by brian-brazil** [comment]
:+1:

**Pull request merged**